src/cloudflare-router.ts

```diff
@@ -1,12 +1,26 @@
-import { Config, DEFAULT_CONFIG } from './config'
+import { Caches } from './binding'
+import { Config } from './config'
 import normalizeRequest from './utils/normalize-request'
 import { withAuth } from './utils/with-auth'
 
+declare let caches: Caches
+
 export const startWorker = (config: Config) => {
   addEventListener('fetch', (event: any) => {
-    withAuth(event, config, () => {
+    withAuth(event, config, async () => {
       const request = normalizeRequest(event.request, config.routes)
-      return fetch(request)
+      // The same cache shared with fetch requests.
+      // Useful when needing to override content that is already cached, after receiving the response.
+      const cache = (caches as Caches).default
+      let response = await cache.match(request.url)
+
+      if (!response) {
+        response = await fetch(request)
+        const headers = { 'cache-control': 'public, max-age=86400' }
```

Comment:
```suggestion
        const headers = {
          ...response.headers,
          'cache-control': 'public, max-age=86400'
        }
```
We should merge the existing headers from the response and only overwrite the ones we need to so we don't accidently remove headers

Reply:
could we include a test that verifies, that we preserve headers, e.g. a `content-type` header that has a value that is not `plain/text` (e.g. `image/svg`)?